packages/bierzo-wallet/src/routes/payment/index.tsx

```diff
@@ -1,87 +1,85 @@
+import { FormValues } from 'medulas-react-components/lib/components/forms/Form';
+import { ToastContext } from 'medulas-react-components/lib/context/ToastProvider';
+import { ToastVariant } from 'medulas-react-components/lib/context/ToastProvider/Toast';
 import React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router';
+import * as ReactRedux from 'react-redux';
 
-import Header from '../../components/Header';
-import CurrencyToSend from './components/CurrencyToSend';
-import ReceiverAddress from './components/ReceiverAddress';
-import TextNote from './components/TextNote';
+import { history } from '..';
+import { sendSignAndPostRequest } from '../../communication/signAndPost';
+//import { sendSignAndPostRequest } from '../../communication/signAndPost';
+import PageMenu from '../../components/PageMenu';
+import { getAddressByName, isIov } from '../../logic/account';
+import { RootState } from '../../store/reducers';
+import { padAmount, stringToAmount } from '../../utils/balances';
+import { PAYMENT_ROUTE } from '../paths';
+import Layout from './components';
+import { CURRENCY_FIELD, QUANTITY_FIELD } from './components/CurrencyToSend';
+import { ADDRESS_FIELD } from './components/ReceiverAddress';
+import { TEXTNOTE_FIELD } from './components/TextNote';
 
-const useStyles = makeStyles((theme: Theme) => ({
-  payment: {
-    backgroundColor: theme.palette.background.default,
-    gridTemplateColumns: '1fr minmax(375px, 450px) 1fr',
-    gridTemplateAreas: `
-  ". currency-to-send ."
-  ". receiver-address ."
-  ". text-note        ."
-  ". continue-button  ."
-  `,
-    gridGap: '24px',
-    placeItems: 'center',
-  },
+function onCancelPayment(): void {
+  history.push(PAYMENT_ROUTE);
+}
 
-  currencyToSend: {
-    gridArea: 'currency-to-send',
-  },
+const Payment = (): JSX.Element => {
+  const toast = React.useContext(ToastContext);
+  const tokens = ReactRedux.useSelector((state: RootState) => state.tokens);
+  const pubKeys = ReactRedux.useSelector((state: RootState) => state.extension.keys);
 
-  receiverAddress: {
-    gridArea: 'receiver-address',
-  },
+  const onSubmit = async (values: object): Promise<void> => {
+    const formValues = values as FormValues;
 
-  textNote: {
-    gridArea: 'text-note',
-  },
+    const ticker = formValues[CURRENCY_FIELD] as TokenTicker;
+    const amount = stringToAmount(formValues[QUANTITY_FIELD], ticker);
+    const token = tokens[ticker];
+    const chainId = token.chainId;
+    const tokenAmount = padAmount(amount, token.token.fractionalDigits);
 
-  continue: {
-    gridArea: 'continue-button',
-  },
-}));
+    let recipient: string | undefined = formValues[ADDRESS_FIELD];
+    if (isIov(recipient)) {
+      recipient = await getAddressByName(recipient.replace(/\*iov$/, ''), chainId);
+    }
 
-const onSubmit = (): void => {};
+    if (!recipient) {
+      toast.show('IOV username was not found', ToastVariant.ERROR);
+      return;
+    }
```

Comment:
This code is very confusing because `recipient` holds the user input as well as the native blockchain address. Could you split that in two variables where the user input can be of type `string | undefined` and the native address of type `Address | undefined`?